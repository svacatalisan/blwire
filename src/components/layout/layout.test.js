import React from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import Layout from './layout';

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

describe('Layout Component', () => {
  let store;
  let component;
  
  beforeEach(() => {
    store = mockStore({
      data: {
        shapes: [],
        filters: [],
          appliedFilters: [] 
      },
    });

    store.dispatch = jest.fn();

    component = create(
      <Provider store={store}>
        <Layout />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
