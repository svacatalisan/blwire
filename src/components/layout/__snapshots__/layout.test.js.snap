// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should render with given state from Redux store 1`] = `
<div
  className="layout-container"
>
  <div
    className="main-header"
  >
    <div
      className="title"
    >
      Shapes
    </div>
  </div>
  <div
    className="home"
  >
    <div
      className="filters"
    >
      <div
        className="literal-filters"
      />
      <div
        className="color-filters"
      />
    </div>
    <div
      className="items-number-container"
    >
      <div
        className="items-number"
      >
        <div
          className="title"
        >
          At least one criterion should be selected from each filter categories
        </div>
      </div>
    </div>
    <div
      className="shapes"
    >
      <div
        className="inline-block"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="shape"
        >
          <div
            style={
              Object {
                "background": undefined,
              }
            }
          />
        </div>
      </div>
      <div
        className="inline-block"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="shape"
        >
          <div
            style={
              Object {
                "background": undefined,
              }
            }
          />
        </div>
      </div>
      <div
        className="inline-block"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="shape"
        >
          <div
            style={
              Object {
                "background": undefined,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  />
</div>
`;
